tests/middleware.test.js

```diff
@@ -0,0 +1,34 @@
+import { expect } from 'chai';
+import errorHandler from '../src/middlewares/errorHandler';
+
+describe('Error Handler', () => {
+  let req, res, next;
+  beforeEach(() => {
+    res = {
+      status(code) {
+        res.status = code;
+      },
+      json() {}
+    };
+    req = {};
+    next = () => {};
+  });
+  it('should return 500 error status', async () => {
+    errorHandler(new Error('Test Error'), req, res, next);
+    expect(res.status).to.be.eql(500);
+  });
+  it('should return specified error for production', async () => {
```

Comment:
```suggestion
it('should return specified error for production', async () => {
    process.env.NODE_ENV = 'production';
    const err = new Error('403 Error'); // won't be there when we define err out the test blocks
    err.status = 403;
 
    errorHandler(err, req, res, next);
 
    expect(res.status).to.be.eql(403);
  });
```
Please, try to do something like this for each block, let the `expect(...)` be after a new line so that it is easy to understand the flow of the tests instead of mushing everything up.

We can use this format for our test blocks;
```
[declarations and assignments] e.g. err.status = 403

[effects i.e. what is actually causing the effect] e.g.  errorHandler(err, req, res, next);

[results] e.g. expect(res.status).to.be.eql(403);
```

You can also define `err` outside the test blocks i.e.

```suggestion
describe('Error Handler', () => {
  let req, res, next;
  const err = new Error('Error');
  ...
```
and then change the error status in each test block. I also think it would be better to use `error` instead of `err`.